src/h5web/vis-packs/core/scalar/utils.ts

```diff
@@ -0,0 +1,18 @@
+import { hasComplexType } from '../../../guards';
+import type {
+  ArrayShape,
+  Dataset,
+  H5WebComplex,
+} from '../../../providers/models';
+import { formatComplex } from '../../../utils';
+import type { PrintableType, ValueFormatter } from '../models';
+
+export function getFormatter(
+  dataset: Dataset<ArrayShape, PrintableType>
+): ValueFormatter<PrintableType> {
+  if (hasComplexType(dataset)) {
+    return (val) => formatComplex(val as H5WebComplex);
```

Comment:
The `getFormatter` for `ScalarVisContainer` -- again, we just cast the value inside the formatter.